assets/app/features/upgrade/steps.factory.js

```diff
@@ -95,5 +98,66 @@
                 }
             ];
         }
+
+        function stepByState(state) {
+            return _.find(factory.steps, function (step) { return step.state === state; });
+        }
+
+        function stepByID(id) {
+            return _.find(factory.steps, function (step) { return step.id === id; });
+        }
+
+        function lastStateForRestore(statusData) {
+            var stepToStateMap = {
```

Comment:
I don't think using this mapping in the unit tests is a good idea as this will hide errors in the mapping itself. The unit tests should explicitly contain input and output values for such cases.